packages/aws-cdk/test/integ/cli/cdk-helpers.ts

```diff
@@ -167,34 +274,43 @@ export class TestFixture {
 }
 
 /**
- * Prepare the app fixture
+ * Perform a one-time quick sanity check that the AWS clients has properly configured credentials
+ *
+ * If we don't do this, calls are going to fail and they'll be retried and everything will take
+ * forever before the user notices a simple misconfiguration.
  *
- * If this is done in the main test script, it will be skipped
- * in the subprocess scripts since the app fixture can just be reused.
+ * We can't check for the presence of environment variables since credentials could come from
+ * anywhere, so do simple account retrieval.
+ *
+ * Only do it once per process.
  */
-export async function prepareAppFixture(env: TestEnvironment): Promise<TestFixture> {
-  const randy = randomString();
-  const stackNamePrefix = `cdktest-${randy}`;
-  const integTestDir = path.join(os.tmpdir(), `cdk-integ-${randy}`);
-
-  env.output.write(` Stack prefixes: ${stackNamePrefix}\n`);
-  env.output.write(` Test directory: ${integTestDir}\n`);
-
-  await cloneDirectory(path.join(__dirname, 'app'), integTestDir, env.output);
-
-  const fixture = new TestFixture(integTestDir, stackNamePrefix, env.output, await AwsClients.default(env.output));
-
-  await fixture.shell(['npm', 'install',
-    '@aws-cdk/core',
-    '@aws-cdk/aws-sns',
-    '@aws-cdk/aws-iam',
-    '@aws-cdk/aws-lambda',
-    '@aws-cdk/aws-ssm',
-    '@aws-cdk/aws-ecr-assets',
-    '@aws-cdk/aws-cloudformation',
-    '@aws-cdk/aws-ec2']);
+async function sanityCheck(aws: AwsClients) {
+  if (sanityChecked === undefined) {
+    try {
+      await aws.account();
+      sanityChecked = true;
+    } catch (e) {
+      sanityChecked = false;
+      throw new Error(`AWS credentials probably not configured, got error: ${e.message}`);
+    }
+  }
+  if (!sanityChecked) {
+    throw new Error('AWS credentials probably not configured, see previous error');
+  }
+}
+let sanityChecked: boolean | undefined;
 
-  return fixture;
+/**
+ * Make sure that the given environment is bootstrapped
+ *
+ * Since we go striping across regions, it's going to suck doing this
+ * by hand so let's just mass-automate it.
+ */
+async function ensureBootstrapped(fixture: TestFixture) {
+  // Old-style bootstrap stack with default name
+  if (await fixture.aws.stackStatus('CDKToolkit') === undefined) {
```

Comment:
Are you sure this is necessary? just running `bootstrap` will no-op if no changes are detected, and if there are changes, shouldn't we bootstrap them?

Reply:
I guess, just cutting down on the duration (since that's the whole goal of this). It's still pretty expensive to do a no-op bootstrap.

Reply:
Ok, i'm still wondering about the need to re-bootstrap when changes are introduced. But thats for a later discussion.